code/__HELPERS/lists.dm

```diff
@@ -137,13 +137,13 @@
 			L[item] = 1
 		total += L[item]
 
-	total = rand(1, total)
+	total = rand()*total
 	for (item in L)
 		total -=L [item]
 		if (total <= 0)
 			return item
 
-	return null
+	return (L && L.len > 0) ? L[L.len] : null
```

Comment:
It took me a pretty good while to figure out what this was even for. It's to catch floating point error. But really it would make more sense to check at the very beginning of the proc if `L` is empty and return then, and just have `return L[L.len]` here, rather than going through the whole proc before validating input.

Alternatively, you could put `. = item` in the second loop, adjacent to `total -= L[item]`, and remove this line entirely, and replace `return item` with just `return`

Or you could leave it as-is. It's up to you. Personally I prefer either of the other two since they're more readable, but all three solutions work.